src/biokbase/narrative/tests/test_exception_to_message.py

```diff
@@ -0,0 +1,313 @@
+            "source": req_type,
+        },
+    }
+
+
+def test_error_with_transform_job_exception(job_comm: JobComm) -> None:
+    """Test the handling of a narrative exception within the context manager using transform_job_exception."""
+    job_id_list = BATCH_CHILDREN
+    req_type = "start_job_status_loop"
+    req = make_job_request(req_type, job_id_list)
+    message = (
+        "In a similar experiment, conducted in Stanford in 1975 ... "
+        "Participants gave the dehumanized people twice the punishment "
+        "of the humanized ones and significantly more than the ones "
+        "they knew nothing about"
+    )
+    err_as_str = str(ValueError(message))
+    error = "Unable to perform this request"
+
+    def transform_exception() -> None:
+        raise transform_job_exception(ValueError(message), error)
+
+    with pytest.raises(NarrativeException, match=message):
+        fn_with_context_manager(transform_exception, req)
+
+    msg = job_comm._comm.last_message
+    assert msg == {
+        "msg_type": ERROR,
+        "content": {
+            "request": req.rq_data,
+            "source": req_type,
+            # Below are from transform_job_exception
+            "name": "ValueError",
+            "message": err_as_str,
+            "error": error,
+            "code": -1,
+        },
+    }
+
+
+def test_error_down_the_stack(job_comm: JobComm) -> None:
+    """Test the handling of an error with some extra stack frames."""
+    job_id = JOB_CREATED
+    req_type = STATUS
+    req_dict = make_comm_msg(req_type, job_id)
+    message = (
+        "Caste is the granting or witholding of "
+        "respect, status, honor, attention, priveliges, "
+        "resources, benefit of the doubt, and human "
+        "kindness to someone on the basis of their "
+        "perceived rank or standing in the hierarchy."
+    )
+
+    # Throw a few frames into stack
+    def add_stack_frames(i: int = 5) -> None:
+        if i == 0:
+            raise ValueError(message)
+        add_stack_frames(i - 1)
+
+    with pytest.raises(ValueError, match=message):
+        fn_with_context_manager(add_stack_frames, req_dict)
+
+    msg = job_comm._comm.last_message
+    assert msg == {
+        "msg_type": ERROR,
+        "content": {
+            "name": "ValueError",
+            "message": message,
+            "request": req_dict["content"]["data"],
+            "source": req_type,
+        },
+    }
+
+
+def test_with_nested_try__raise(job_comm: JobComm) -> None:
+    """Test the handling of an exception that triggers the try/except handler."""
+    job_id_list = [BATCH_TERMINATED, JOB_TERMINATED]
+    req_type = "stop_job_status_loop"
+    req = make_job_request(req_type, job_id_list)
+    message = (
+        "Casteism is the investment in keeping the hierarchy "
+        "as it is in order to maintain your own ranking, advantage, "
+        "privilege, or to elevate yourself above others or keep "
+        "others beneath you"
```

Comment:
I feel like these should really be cited properly so no one thinks KBase developers have such deep, philosophical thoughts whilst working on the narrative codebase.

Reply:
I somehow don't think others will really get that impression...